docs/dev/dev_setup.rst

```diff
@@ -203,11 +203,97 @@ with linters and other code analysis tools.
+    skip ahead to :ref:`Legacy PUDL Setup`. If you haven't setup
+    a PUDL workspace before, read the remainder of this section.
+
+PUDL needs to know where to store its big piles of inputs and outputs.
+The ``PUDL_OUTPUT`` and ``PUDL_CACHE`` environment variables let PUDL know where
+all this stuff should go. We call this a "PUDL workspace".
+
+First, create a directory to store local caches of raw PUDL data. Typically we
+put this in ``~/pudl-work/data``. Then create an environment variable
+called ``PUDL_CACHE`` to store the path to this new directory:
+
+.. code-block:: console
+
+    $ echo "export PUDL_CACHE=/path/to/cache/dir" >> ~/.zshrc # zsh
+    $ echo "export PUDL_CACHE=/path/to/cache/dir" >> ~/.bashrc # bash
+    $ set -Ux PUDL_CACHE /path/to/cache/dir # fish
+
+The directory stored in ``PUDL_CACHE`` contains versions of PUDL's
+raw data archives on Zenodo for each datasource:
+
+.. code-block::
+
+    pudl_cache/
+    ├── ferc1/
+    │   ├── 10.5281-zenodo.5534788/
+    │   │   ├── datapackage.json
+    │   │   ├── ferc1-1994.zip
+    │   │   ├── ferc1-1995.zip
+    │   │   └── ...
+    │   ├── 10.5281-zenodo.7314437/
+    │   │   └── ...
+    │   └── ...
+    ├── eia860/
+    │   └── ...
+    └── ...
+
+.. warning::
+
+    The data stored at the ``PUDL_CACHE`` directory can grow to be dozens
+    of gigabytes in size. This is because when the raw data are updated,
+    a new version of the archive is downloaded to the ``PUDL_CACHE``
+    directory. To slim down the size you can always delete
+    out of date archives the code no longer depends on.
+
+Next, create a directory to store the outputs of the PUDL ETL. Typically this
+is ``~/pudl-work/output``. Then, as with ``PUDL_CACHE``, create an environment
+variable called ``PUDL_OUTPUT`` to store the path to this new directory:
+
+.. code-block:: console
+
+    $ echo "export PUDL_OUTPUT=/path/to/output/dir" >> ~/.zshrc # zsh
+    $ echo "export PUDL_OUTPUT=/path/to/output/dir" >> ~/.bashrc # bash
+    $ set -Ux PUDL_OUTPUT /path/to/output/dir # fish
+
+The path stored in ``PUDL_OUTPUT`` contains all ETL outputs like
+``pudl.sqlite`` and ``hourly_emissions_epacems.parquet``.
+
+**Make sure you create separate directories for these environment variables!
+It is recommended you create these directories outside of the pudl repository
+directory so the inputs and outputs are not tracked in git.**
+
+.. _Legacy PUDL Setup:
+
+PUDL Workspace Setup (legacy method)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In previous versions of PUDL, the ``pudl_setup`` script created workspace directories.
+PUDL is moving towards using the ``PUDL_OUTPUT`` and ``PUDL_CACHE`` environment
+variables instead of the ``pudl_setup`` script because the environment variables are
+easier to reference in the codebase.
+
+.. note::
+
+    If you set up your workspace using ``pudl_setup`` you don't need to change
+    anything about your setup. Just re-run ``pudl_setup`` and a new directory
```

Comment:
Maybe it's just me, but running pudl_setup on my existing legacy version did not add a new directory as indicated here. Is this an error, or is the functionality noted here not actually implemented in the code?

Reply:
I think this is an error - we used to do this and now it doesn't happen.

Reply:
But we can just... fix that.